### PR TITLE
Support OpenGauss PREDICT BY function parse

### LINKED ISSUE #27652: Support parsing openGauss  SELECT sql
# Background
Hi community,

ShardingSphere parser engine helps users parse a SQL to get the AST (Abstract Syntax Tree) and visit this tree to get SQLStatement (Java Object). Currently, we are planning to enhance the support for openGauss SQL parsing in ShardingSphere.

More details:
https://shardingsphere.apache.org/document/current/en/reference/sharding/parse/

# Task
This issue is to support more openGauss sql parse, as follows:
```sql
 SELECT id, PREDICT BY point_kmeans (FEATURES position) as pos FROM (select * from kmeans_2d limit 10);
```

```sql
 Explain CREATE MODEL patient_logisitic_regression USING logistic_regression FEATURES second_attack, treatment TARGET trait_anxiety > 50 FROM patients WITH batch_size=10, learning_rate = 0.05;
```

```sql
 CREATE MODEL patient_linear_regression USING linear_regression FEATURES second_attack,treatment TARGET trait_anxiety  FROM patients WITH optimizer='aa';
```

```sql
 CREATE MODEL patient_linear_regression USING linear_regression FEATURES second_attack,treatment TARGET trait_anxiety  FROM patients;
```

```sql
 select id, PREDICT BY patient_logistic_regression (FEATURES second_attack,treatment) FROM patients;
```

# Process
1. First confirm that this is a correct openGauss sql syntax, if not please leave a message under the issue and ignore it;
2. Compare SQL definitions in Official SQL Doc and ShardingSphere SQL Doc;
3. If there is any difference in ShardingSphere SQL Doc, please correct them by referring to the Official SQL Doc;
4. Run mvn install the current_file_module;
5. Check whether there are any exceptions. If indeed, please fix them. (Especially xxxVisitor.class);
6. Add new corresponding SQL case in SQL Cases and expected parsed result in Expected Statement XML;
7. Run SQLParserParameterizedTest to make sure no exceptions.

# Relevant Skills
1. Master JAVA language
2. Have a basic understanding of Antlr `g4` file
3. Be familiar with openGauss SQLs

### PR DESCRIPTION
**Pull request by Qianyi951015**
Fixes #27652.

Changes proposed in this pull request:
  - Support OpenGauss PREDICT BY function parse

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: parser/sql/dialect/opengauss/src/main/antlr4/imports/opengauss/BaseRule.g4 [hunk 1]
```diff
@@ -494,6 +495,10 @@ name
     : identifier
     ;
 
+modelName
```
Comment (strongduanmu) [inline parser/sql/dialect/opengauss/src/main/antlr4/imports/opengauss/BaseRule.g4:498]: Can you add test case for this statement?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: parser/sql/dialect/opengauss/src/main/antlr4/imports/opengauss/BaseRule.g4 (hunk 1 above)
Comment (Qianyi951015) [reply parser/sql/dialect/opengauss/src/main/antlr4/imports/opengauss/BaseRule.g4:498]: sure

**Review by FlyingZC — APPROVED** [review]

**Review by strongduanmu — APPROVED** [review]
